### DISCUSSION TITLE
Token claim `typ` invalid

### DISCUSSION THREAD

**Discussion by lcdunne** [keycloak/keycloak#32216] [Q&A]
Hi. I am trying to validate a token obtained from my Keycloak server but I was receiving `invalid_claim: Invalid claim "typ"`. After digging into it, I see that it is due to [RFC9068](https://www.rfc-editor.org/rfc/rfc9068.html#name-validating-jwt-access-token), which specifies that it (nowadays) should be either `at+jwt` or `application/at+jwt`, but not `jwt` as is the case with Keycloak. The library I am using is obviously trying to keep with current standards so I guess it doesn't support `JWT` anymore. Is there a way to change this so that Keycloak generates tokens that have `typ=at+jwt`? If not, is there a plan to update this? Thanks.

**Comment by codespearhead** [comment]
I suggest [closing this discussion](https://docs.github.com/en/discussions/managing-discussions-for-your-community/managing-discussions#closing-a-discussion) in favor of https://github.com/keycloak/keycloak/issues/19220 and giving that issue a thumbs-up. Here's an example implementation in case you're willing to work on a PR: https://github.com/keycloak/keycloak/pull/29260 .

**Comment by laurids** [comment]
Hi @lcdunne 
From version 26.2, there will be a way to set the access token header type "at+jwt" in the per-client configuration. See https://github.com/keycloak/keycloak/pull/37573
